wfc_gh/WaveFunctionCollapse.cs

```diff
@@ -464,20 +464,27 @@ protected override void SolveInstance(IGH_DataAccess DA)
             bool foundDeterministic = false;
             uint attempts = 0;
 
-            while (!foundDeterministic && attempts < maxAttempts)
-            {
-                var result = Native.wfc_observe(wfcWorldStateHandle, wfcRngStateHandle);
-                if (result == WfcObserveResult.Deterministic)
+            uint maxObservations = UInt32.MaxValue;
+            uint spentObservations = 0;
+            
+            unsafe{
+                while (!foundDeterministic && attempts < maxAttempts)
                 {
-                    foundDeterministic = true;
-                }
-                else
-                {
-                    Native.wfc_world_state_clone_from(wfcWorldStateHandle,
-                                                      wfcWorldStateHandleBackup);
+                    var result = Native.wfc_observe(wfcWorldStateHandle,
+                                                    wfcRngStateHandle,
+                                                    (UIntPtr)maxObservations,
+                                                    &spentObservations);
+                    if (result == WfcObserveResult.Deterministic)
+                    {
+                        foundDeterministic = true;
+                    }
+                    else
+                    {
+                        Native.wfc_world_state_clone_from(wfcWorldStateHandle,
+                                                        wfcWorldStateHandleBackup);
+                    }
```

Comment:
Should this be rolled back if all observations were spent?

Reply:
Yeah. The entire GH component is a bit outdated.
I've hidden the step-solver features so that it works like before.